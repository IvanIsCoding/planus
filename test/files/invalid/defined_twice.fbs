struct Foo {}
table Foo {}


struct A0 {
	x: uint32;
	x: string;
}

table A1 {
	x: uint32;
	x: string;
}

enum A2: uint8 {
 X = 0, X = 1
}

union A3 {
 Foo, Foo
}

union A4 {
 X: A0, X: A1
}

rpc_service A5 {
  x(string): string;
  x(uint32): uint32;
}
