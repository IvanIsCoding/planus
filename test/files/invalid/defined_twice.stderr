[0m[1m[38;5;9merror[0m[1m: cannot define type Foo twice[0m
  [0m[34mâ”Œâ”€[0m test/files/invalid/defined_twice.fbs:1:1
  [0m[34mâ”‚[0m
[0m[34m1[0m [0m[34mâ”‚[0m table Foo {}
  [0m[34mâ”‚[0m [0m[34m------------[0m [0m[34mfirst definition was here[0m
[0m[34m2[0m [0m[34mâ”‚[0m union Foo {}
  [0m[34mâ”‚[0m [0m[34m------------[0m [0m[34msecond definition was here[0m

[0m[1m[38;5;9merror[0m[1m: cannot define field x twice[0m
  [0m[34mâ”Œâ”€[0m test/files/invalid/defined_twice.fbs:7:3
  [0m[34mâ”‚[0m
[0m[34m7[0m [0m[34mâ”‚[0m   x: uint32;
  [0m[34mâ”‚[0m   [0m[34m----------[0m [0m[34mfirst field was here[0m
[0m[34m8[0m [0m[34mâ”‚[0m   x: string;
  [0m[34mâ”‚[0m   [0m[34m----------[0m [0m[34msecond field was here[0m

[0m[1m[38;5;9merror[0m[1m: cannot define field x twice[0m
   [0m[34mâ”Œâ”€[0m test/files/invalid/defined_twice.fbs:12:3
   [0m[34mâ”‚[0m
[0m[34m12[0m [0m[34mâ”‚[0m   x: uint32;
   [0m[34mâ”‚[0m   [0m[34m----------[0m [0m[34mfirst field was here[0m
[0m[34m13[0m [0m[34mâ”‚[0m   x: string;
   [0m[34mâ”‚[0m   [0m[34m----------[0m [0m[34msecond field was here[0m

[0m[1m[38;5;9merror[0m[1m: cannot define variant X twice[0m
   [0m[34mâ”Œâ”€[0m test/files/invalid/defined_twice.fbs:17:3
   [0m[34mâ”‚[0m
[0m[34m17[0m [0m[34mâ”‚[0m   X = 0, X = 1
   [0m[34mâ”‚[0m   [0m[34m------[0m [0m[34m-----[0m [0m[34msecond variant was here[0m
   [0m[34mâ”‚[0m   [0m[34mâ”‚[0m       
   [0m[34mâ”‚[0m   [0m[34mfirst variant was here[0m

[0m[1m[38;5;9merror[0m[1m: cannot define union variant with type Foo twice[0m
   [0m[34mâ”Œâ”€[0m test/files/invalid/defined_twice.fbs:21:3
   [0m[34mâ”‚[0m
[0m[34m21[0m [0m[34mâ”‚[0m   Foo, Foo
   [0m[34mâ”‚[0m   [0m[34m----[0m [0m[34m---[0m [0m[34msecond variant was here[0m
   [0m[34mâ”‚[0m   [0m[34mâ”‚[0m     
   [0m[34mâ”‚[0m   [0m[34mfirst variant was here[0m

[0m[1m[38;5;9merror[0m[1m: cannot define union variant with name X twice[0m
   [0m[34mâ”Œâ”€[0m test/files/invalid/defined_twice.fbs:25:3
   [0m[34mâ”‚[0m
[0m[34m25[0m [0m[34mâ”‚[0m   X: Foo, X: Bar
   [0m[34mâ”‚[0m   [0m[34m-------[0m [0m[34m------[0m [0m[34msecond variant was here[0m
   [0m[34mâ”‚[0m   [0m[34mâ”‚[0m        
   [0m[34mâ”‚[0m   [0m[34mfirst variant was here[0m

[0m[1m[38;5;9merror[0m[1m: cannot define rpc method x twice[0m
   [0m[34mâ”Œâ”€[0m test/files/invalid/defined_twice.fbs:29:3
   [0m[34mâ”‚[0m
[0m[34m29[0m [0m[34mâ”‚[0m   x(string): string;
   [0m[34mâ”‚[0m   [0m[34m------------------[0m [0m[34mfirst method was here[0m
[0m[34m30[0m [0m[34mâ”‚[0m   x(uint32): uint32;
   [0m[34mâ”‚[0m   [0m[34m------------------[0m [0m[34msecond method was here[0m

[0m[1m[38;5;9merror[0m[1m: Rpc services are not currently supported[0m
   [0m[34mâ”Œâ”€[0m test/files/invalid/defined_twice.fbs:28:1
   [0m[34mâ”‚[0m
[0m[34m28[0m [0m[34mâ”‚[0m [0m[31mrpc_service A5[0m {
   [0m[34mâ”‚[0m [0m[31m^^^^^^^^^^^^^^[0m

Error: Bailing because of errors
